packages/default-theme/components/SwCurrency.vue

```diff
@@ -0,0 +1,93 @@
+<template>
+  <div class="container" v-if="activeCurrency && availableCurrencies && availableCurrencies.length">
+    <SfSelect v-model="activeCurrency" :size="availableCurrencies.length" class="container__select">
+      <SfSelectOption
+        v-for="currencyItem in availableCurrencies"
+        :key="currencyItem.id"
+        :value="currencyItem.id"
+      >
+        {{currencyItem.symbol}}
+      </SfSelectOption>
+    </SfSelect>
+  </div>
+</template>
+<script>
+import { SfSelect, SfProductOption } from '@storefront-ui/vue';
+import { useCurrency } from '@shopware-pwa/composables';
+import { computed } from '@vue/composition-api';
+
+export default {
+  name: 'SwCurrency',
+  components: {
+    SfSelect
+  },
+  setup() {
+    const { 
+      availableCurrencies,
+      currentCurrency,
+      fetchCurrencies,
+      switchCurrency
+      } = useCurrency()
+
+    const activeCurrency = computed({
+      get: () => currentCurrency.value && currentCurrency.value.id,
+      set: (currencyId) => switchCurrency(currencyId).then(() => {
+        if (window) {
```

Comment:
maybe we could try with `this.$router.go()`
https://stackoverflow.com/questions/41301099/do-we-have-router-reload-in-vue-router

Reply:
this.$router.go needs to be this aware, but it's not inside nested callbacks